### ISSUE TITLE
[java] CloseResource: false positive violation with closeNotInFinally=true property

### ISSUE THREAD

**Issue by dykov** [pmd/pmd#3586]
**PMD version:** 6.39.0
**Rule:** [CloseResource](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#closeresource)
**Rule in ruleset:**
```xml
  <rule ref="rulesets/java/design.xml/CloseResource">
    <properties>
      <property name="closeNotInFinally" value="true" />
    </properties>
  </rule>
```

**Code to reproduce:**
```java
package com.example;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class SomeClass {
    private void method() throws SQLException {
        Connection connection = DriverManager.getConnection("/*...*/");   //violation
        connection.close();
    }
}
```

Please check if these functions work correctly?
net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule#visit:
```java
    @Override
    public Object visit(ASTPrimaryPrefix prefix, Object data) {
        //...
        ASTName methodCall = prefix.getFirstChildOfType(ASTName.class);
        if (methodCall != null && isNodeInstanceOfResourceType(methodCall)) {
       //...
```

net.sourceforge.pmd.lang.java.rule.errorprone.CloseResourceRule#isNodeInstanceOfResourceType:
```java
private boolean isNodeInstanceOfResourceType(TypeNode refType) {
        for (String resType : types) {
            if (TypeTestUtil.isA(resType, refType)) {
                return true;
            }
        }
        return false;
    }
```
So, we provide `ASTName` object with a name of invocation (e.g. `connection.close()`) to the `TypeTestUtil#isA`. 
Is it allowed? As I understand, `TypeTestUtil#isA` should receive `ASTClassOrInterfaceType` object and the canonical name of the class to compare?

Thank you

**Comment by dykov** [comment]
Thanks a lot for your help!
